### PR TITLE
Fix debug messages when rBoot is enabled

### PR DESCRIPTION
**Pull request by gbl08ma**
Apparently, the people in charge of #927 forgot to apply the changes necessary for things to work on ROMs compiled with rBoot in mind. This resulted in all Sming debug messages appearing as garbage on such ROMs:

garbage debug messages

This pull request fixes the issue by applying the necessary changes to the rBoot-specific Makefile and also to the example rBoot linker script.

In addition, I changed some things in the rBoot-specific Makefile to make it more similar to the common one. I did not port the new memory information output as it seemed to require quite a bit more effort than my time allowed, and I guess that can be postponed to a later pull request...

This has been tested and confirmed to work for my projects only. I can't guarantee that other configurations will still work, but I don't see why not.

### CODE REVIEW

**Comment by ADiea** [comment]
Good job and sorry for the missing code!

File: Sming/Makefile-rboot.mk [hunk 1]
```diff
@@ -248,11 +253,6 @@ ifeq ($(ENABLE_SSL),1)
 	CXXFLAGS += $(AXTLS_FLAGS)	
 endif
 
-ifeq ($(ENABLE_CUSTOM_LWIP), 1)
```
Comment (slaff) [inline Sming/Makefile-rboot.mk]: Why is that remove?!

File: Sming/Makefile-rboot.mk [hunk 2]
```diff
@@ -173,31 +179,7 @@ endif
 
 EXTRA_INCDIR += $(SMING_HOME)/include $(SMING_HOME)/ $(LWIP_INCDIR) $(SMING_HOME)/system/include $(SMING_HOME)/Wiring $(SMING_HOME)/Libraries $(SMING_HOME)/SmingCore $(SMING_HOME)/Services/SpifFS $(SDK_BASE)/../include $(THIRD_PARTY_DIR)/rboot $(THIRD_PARTY_DIR)/rboot/appcode $(THIRD_PARTY_DIR)/spiffs/src
 
-USER_LIBDIR  = $(SMING_HOME)/compiler/lib/
```
Comment (slaff) [inline Sming/Makefile-rboot.mk]: Why was that paragraph placed in another location?!

File: Sming/Makefile-rboot.mk [hunk 3]
```diff
@@ -153,8 +159,8 @@ TARGET		= app
 
 THIRD_PARTY_DIR = $(SMING_HOME)/third-party
 
-LIBSMING = sming
 SMING_FEATURES = none
+LIBSMING = sming
```
Comment (slaff) [inline Sming/Makefile-rboot.mk]: Why is that change here?!

Comment (gbl08ma) [inline Sming/Makefile-rboot.mk]: "In addition, I changed some things in the rBoot-specific Makefile to make it more similar to the common one."
If you looked at the diff between `Makefile-project.mk` and `Makefile-rboot.mk` before and after this commit, instead of simply looking at the commit diff, perhaps you would understand the end result better.

By reducing the amount of differences between these two files, it becomes easier to correctly apply changes to both at the same time, and perhaps even merge parts of them into a single file one day.

File: Sming/Makefile-rboot.mk (hunk 2 above)
Comment (gbl08ma) [inline Sming/Makefile-rboot.mk]: I simply moved it to the same place where it was in `Makefile-project.mk`, but on second analysis it seems that perhaps its location in `Makefile-project.mk` is wrong too, since I see references to `USER_LIBDIR` before it is set. Furthermore, now there's a duplicate line, `# libraries used in this project, mainly provided by the SDK`, which wasn't there before. I'll fix this and move the `USER_LIBDIR` line back to before all its usages, but we should check if `Makefile-project.mk` isn't wrong too.

File: Sming/Makefile-rboot.mk (hunk 1 above)
Comment (gbl08ma) [inline Sming/Makefile-rboot.mk]: Because `LWIP_INCDIR` is already being added to `EXTRA_INCDIR` on line 180 of my `Makefile-rboot.mk` (or line 174 of yours). While apparently the path is not the same (it doesn't end in `/include`), if you look at `Makefile-project.mk`, you'll see that these lines are not there, either, and only the path that doesn't end in `include` is ever added to `EXTRA_INCDIR` (or is it wrong too? doubt so, as that would have broken custom LWIP for everyone, I think).

Furthermore, the path it was adding to `EXTRA_INCDIR` was missing the `$(SMING_HOME)` prefix and I doubt it would work when the makefile is run from the project directory.

File: Sming/Makefile-rboot.mk (hunk 2 above)
Comment (slaff) [inline Sming/Makefile-rboot.mk]: @gbl08ma Please, make only one logical change per PR. It makes things easier for us to review and maintain the code. If there are other issues in the makefile(s) then make a second, third, etc PR. Therefore I would like to ask you to remove all changes that are not related to "Fix debug messages when rBoot is enabled". If you cannot do it then I will have to rebase this PR later this week.

Comment (gbl08ma) [inline Sming/Makefile-rboot.mk]: The reason why there are all these changes is because I was having trouble bringing in the changes related to the debug messages with the rBoot makefile being so different from the default one. So I basically had to diff them and go change by change applying (or not) each one.

I wouldn't mind trying again but this requires testing on the hardware to see if it's fixed, and I no longer have the ESP8266 I was using with me, so this may have to wait some days.

**Comment by slaff** [comment]
... Will be merged with #947. Thanks for noticing the issues and reporting them.

**Comment by anakod** [comment]
@gbl08ma 
> By reducing the amount of differences between these two files, it becomes easier to correctly apply changes to both at the same time, and perhaps even merge parts of them into a single file one day.

May be we can go step forward and use single Makefile with different options for rboot and usual build?

**Comment by ADiea** [comment]
like ENABLE_REBOOT=1 ? or it's not that simple? Never used rboot until now but planning to...
